specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json

```diff
@@ -928,6 +1007,218 @@
         },
         "upgradeDetails": {
           "$ref": "#/definitions/UpgradeDetails"
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnectionVaultProperties"
+          },
+          "description": "List of private endpoint connection.",
+          "readOnly": true
+        },
+        "privateEndpointStateForBackup": {
+          "description": "Private endpoint state for backup.",
+          "enum": [
+            "None",
+            "Enabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "VaultPrivateEndpointState",
+            "modelAsString": true
+          }
+        },
+        "privateEndpointStateForSiteRecovery": {
+          "description": "Private endpoint state for site recovery.",
+          "enum": [
+            "None",
+            "Enabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "VaultPrivateEndpointState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "IdentityData": {
+      "required": [
+        "type"
+      ],
+      "type": "object",
+      "description": "Identity for the resource.",
+      "properties": {
+        "principalId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The principal ID of resource identity."
+        },
+        "tenantId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The tenant ID of resource."
+        },
+        "type": {
+          "type": "string",
+          "description": "The identity type.",
+          "enum": [
+            "SystemAssigned",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "ResourceIdentityType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "PrivateEndpointConnectionVaultProperties": {
+      "description": "Information to be stored in Vault properties as an element of privateEndpointConnections List. It is to be returned to RRP from BMS as response of PUT PECP and PUT PEC calls.",
+      "readOnly": true,
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Format of id subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.[Service]/{resource}/{resourceName}/privateEndpointConnections/{connectionName}.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/PrivateEndpointConnection"
+        }
+      }
+    },
+    "PrivateEndpointConnection": {
```

Comment:
>"PrivateEndpointConnection" [](start = 4, length = 27)

you should have a child resource called privateEndpointConnections that exposes the full set of private endpoint properties.  See other RPs (i.e. cosmos db, key vault) for examples

Reply:
We already  have a property called PrivateEndpointConnections which exposes a full list of PrivateEndpointConnectionVaultProperties. This PrivateEndpointConnectionVaultProperties internally contains two properties : PrivateEndpointConnection and id.For each PrivateEndpointConnection’s properties, we contain only the id of the privateEndpoint as specified in section 6.5.2 of  NRP Private Endpoint Design Doc (https://microsoft.sharepoint.com/:w:/r/teams/WAG/AzureNetworking/_layouts/15/Doc.aspx?sourcedoc=%7B64F10150-563A-4715-9476-FFA24F9BD42E%7D&file=%5BNRP%5D%20Private%20Endpoint%20Design%203.docx&wdLOR=cC8C98543-7B1C-470B-A557-E1E3501F6F9B&action=default&mobileredirect=true)

Reply:
Please follow the example set by other RPs.  This part is correct but PrivateEndpointConnections should also be exposed as a child resource.  You should never have something with a resource ID and provisioningState that doesn't have a corresponding GET at that resource ID (and a GET collection for all of that type at the scope).  See cosmos DB, key vault, etc... for an example

Reply:
Hi, we have added GET for this PrivateEndpointConnection resource. Please check the below PR: 
https://github.com/Azure/azure-rest-api-specs/pull/8587